File: 204-count-primes/count-primes.py
class Solution:
    def countPrimes(self, n: int) -> int:
        # divisible by 1 and itself,
        # brute force check each number
        seen, ans = [0]*n, 0
        for num in range(2,n):
            if seen[num]: continue
            ans+=1
            seen[num*num:n:num] =[1] *((n-1) // num-num+1)
        return ans

        

File: 204-count-primes/count-primes.cpp
class Solution {
public:
    int countPrimes(int n) {
        vector<bool> seen(n, false);
        int ans = 0;
        for (int num = 2; num < n; num++) {
            if (seen[num])
                continue;
            ans++;
            for (long m = (long)num * num; m < n; m += num) {
                seen[m] = true;
            }
        }
        return ans;
    }
};